v3.1/glfw/util.go

```diff
@@ -1,11 +1,36 @@
 package glfw
 
+//#include <stdlib.h>
 //#include "glfw/include/GLFW/glfw3.h"
 import "C"
 
+import (
+	"reflect"
+	"unsafe"
+)
+
 func glfwbool(b C.int) bool {
 	if b == C.GL_TRUE {
 		return true
 	}
 	return false
 }
+
+func bytes(origin []byte) (pointer *uint8, free func()) {
+	if len(origin) == 0 {
+		panic("Slice of bytes cannot be empty.")
+	}
+
+	l := len(origin)
+	d := C.malloc(C.size_t(l))
+
+	ds := *(*[]byte)(unsafe.Pointer(&reflect.SliceHeader{
```

Comment:
At first I found it hard to understand what `d` and `ds` stood for, then I realized it was a short version of `data` and `dataSlice` from original `gl.Strs` code. I think the original names were more readable, let's stick with those, unless you feel strongly about it.